source/MaterialXTest/ShaderValid.cpp

```diff
@@ -449,99 +407,86 @@ TEST_CASE("GLSL MaterialX documents", "[shadervalid]")
 
             std::vector<mx::NodeGraphPtr> nodeGraphs = doc->getNodeGraphs();
             std::vector<mx::OutputPtr> outputList = doc->getOutputs();
-            // TO ADD: Checking for shaders to validate
-            if (!nodeGraphs.empty() || !outputList.empty())
+            std::unordered_set<mx::OutputPtr> outputSet(outputList.begin(), outputList.end());
+            std::vector<mx::MaterialPtr> materials = doc->getMaterials();
+
+            if (!materials.empty() || !nodeGraphs.empty() || !outputList.empty())
             {
                 log << "MTLX Filename: " << filename << std::endl;
+
+                std::unordered_set<mx::OutputPtr> shaderrefOutputs;
+                for (auto material : materials)
+                {
+                    for (auto shaderRef : material->getShaderRefs())
+                    {
+                        if (!shaderRef->hasSourceUri())
+                        {
+                            mx::HwShaderGenerator& hwGenerator = static_cast<mx::HwShaderGenerator&>(*glslShaderGenerator);
+
+                            // Set up lighting
+                            mx::HwLightHandlerPtr lightHandler = mx::HwLightHandler::create();
+                            createLightRig(doc, *lightHandler, hwGenerator);
```

Comment:
Why is the light rig re-created for each loop iteration, why not more it out from the loop and only create the lights once?